modules/gdpr_fields/src/Plugin/Deriver/TypedDataEntityRelationshipReverseDeriver.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Drupal\gdpr_fields\Plugin\Deriver;
+
+use Drupal\Core\Plugin\Context\ContextDefinition;
+use Drupal\Core\TypedData\DataDefinitionInterface;
+use Drupal\ctools\Plugin\Deriver\TypedDataEntityRelationshipDeriver;
+
+/**
+ * Derives reverse entity relationships.
+ */
+class TypedDataEntityRelationshipReverseDeriver extends TypedDataEntityRelationshipDeriver {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $label = '@property Entity referencing @base';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function generateDerivativeDefinition($base_plugin_definition, $data_type_id, $data_type_definition, DataDefinitionInterface $base_definition, $property_name, DataDefinitionInterface $property_definition) {
+    if (method_exists($property_definition, 'getType') && $property_definition->getType() == 'entity_reference') {
+      parent::generateDerivativeDefinition($base_plugin_definition, $data_type_id, $data_type_definition, $base_definition, $property_name, $property_definition);
+
+      // Provide the entity type.
+      $derivative_id = $data_type_id . ':' . $property_name;
+      if (isset($this->derivatives[$derivative_id])) {
```

Comment:
This throws notices and results in a "function call on null" error down the line, as 'target_entity_type' doesn't exist.

Using `if (isset($this->derivatives[$derivative_id]['target_entity_type'])) {` solves the problem.

Reply:
No, if the target_entity_type is not set we have a problem anyway. This is a required setting. I think the problem is more in the building the right derivative id. My change should have fixed this. Can you confirm?